core/src/main/java/org/apache/iceberg/encryption/EncryptionManagerFactory.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.encryption;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.iceberg.TableMetadata;
+
+public interface EncryptionManagerFactory extends Closeable {
+
+  EncryptionManagerFactory NO_ENCRYPTION = tableMetadata -> PlaintextEncryptionManager.instance();
+
+  /**
+   * Initialize EncryptionManagerFactory from catalog properties.
+   *
+   * @param properties catalog properties
+   */
+  default void initialize(Map<String, String> properties) {}
+
+  /**
+   * Create encryption manager from table metadata.
+   *
+   * @param tableMetadata table metadata
+   * @return created encryption manager instance.
+   */
+  EncryptionManager create(TableMetadata tableMetadata);
```

Comment:
This is over-broad. If the idea is to create an encryption manager from a table's properties, then it should accept only properties.

Reply:
when we get to column encryption, there is a chance we might need the schema here. Not sure yet, but the reason for passing the metadata was to get the schema from it in the future.

Reply:
Per our recent discussion, will send properties only for now. Later, we'll modify the API to add column-related metadata as needed.